### ISSUE TITLE
LearningRateMonitor(logging_interval='epoch',...) should plot learning-rate versus epochs as opposed to learning-rate versus steps

### ISSUE THREAD

**Issue by vineetk1** [PyTorchLightning/pytorch-lightning#5594]
## 🚀 Feature
LearningRateMonitor(logging_interval='epoch',...) should plot learning-rate versus epochs as opposed to learning-rate versus steps

### Motivation
Since I am specifying that the logging-interval is in epochs, I expect TensorBoard to show the x-axis in epochs as opposed to steps.

### Additional context
I have made the following change in line 131 of file pytorch_lightning/callbacks/lr_monitor.py:
Replace:
`trainer.logger.log_metrics(latest_stat, step=trainer.global_step)`
with:
`trainer.logger.log_metrics(latest_stat, step=trainer.current_epoch)`
This change in code works. I am using Pytorch-Lighting version 1.1.4

**Comment by rohitgr7** [comment]
let's decide it finally :sweat_smile: ... what should be there.. `epoch` or `global_step`? because it's already changed twice now AFAIK. cc @PyTorchLightning/core-contributors

**Comment by SkafteNicki** [comment]
Here is the issue that requested the current behaviour: https://github.com/PyTorchLightning/pytorch-lightning/issues/4375.
I think this is not unique to the learning rate monitor. For every value we log the x-axis is always shown in steps and we provide the epoch graph for conversion.

**Comment by awaelchli** [comment]
The abscissa should show the global step. I strongly advise against changing this behaviour. Working with the global step is the best and most accurate way we currently have to compare experiments. Changing the interval to epochs should just log less often (only each epoch) but still have the global step as the reference.

**Comment by williamFalcon** [comment]
i agree here. global_step is the only way to compare across everything. 

please keep global_step

**Comment by vineetk1** [comment]
I think we could distinguish between "logging_interval" and the units of the x-axis. That would mean another parameter through which a user can specify the units of the x-axis. The options could be: example, mini-batch, epoch.
If I am plotting graphs with different batch-sizes then I would want my x-axis in the units of "examples".
Just my 2 cents.
Feel free to close this issue, since there is a consensus against making changes.

**Comment by rohitgr7** [comment]
you can still override `on_train_epoch_start` hook in `LearningRateMonitor:
```python
class CustomLearningRateMonitor(LearningRateMonitor):
    def on_train_epoch_start(self, trainer, *args, **kwargs):
        ....
        trainer.logger.log_metrics(..., step=trainer.current_epoch)
```
you can check the codebase for the rest of the code (just 3-4 more LOC are there).
